debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/transforms/ReadToInsertEvent.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mysql.transforms;
+
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Configuration;
+import io.debezium.data.Envelope;
+import io.debezium.transforms.SmtManager;
+
+public class ReadToInsertEvent<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ReadToInsertEvent.class);
+
+    private SmtManager<R> smtManager;
+
+    @Override
+    public R apply(R record) {
+        if (record.value() == null || !smtManager.isValidEnvelope(record)) {
+            return record;
+        }
+
+        Struct originalValueStruct = (Struct) record.value();
+        Struct updatedValueStruct;
+        String operation = originalValueStruct.getString(Envelope.FieldName.OPERATION);
+
+        if (operation.equals(Envelope.Operation.READ.code())) {
+            updatedValueStruct = updatedValue(originalValueStruct);
+        }
+        else {
+            return record;
+        }
+
+        return record.newRecord(
+                record.topic(),
+                record.kafkaPartition(),
+                record.keySchema(),
+                record.key(),
+                record.valueSchema(),
+                updatedValueStruct,
+                record.timestamp());
+    }
+
+    private Struct updatedValue(Struct originalValue) {
+        Struct updatedValue = new Struct(originalValue.schema());
+        for (org.apache.kafka.connect.data.Field field : originalValue.schema().fields()) {
```

Comment:
I might be missing something, but couldn't you effectively just do this:

    originalValue.put("op", Envelope.Operation.CREATE.code())

And be done with it? Why the need for that iteration?

Reply:
Ah, my bad. I thought in case there won't be `op` field it would be necessary to run a check, hence the rechecking. But `op` field is a must.

Reply:
Yes, for actual change data events, "op" is always present. And you do have the check for the envelope being present somewhere before.